services/dedup/mirrorBadger/mirrorBadger.go

```diff
@@ -0,0 +1,40 @@
+package mirrorBadger
+
+import (
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/services/dedup/badger"
+	"github.com/rudderlabs/rudder-server/services/dedup/scylla"
+	"github.com/rudderlabs/rudder-server/services/dedup/types"
+)
+
+type MirrorBadger struct {
+	badger *badger.Dedup
+	scylla *scylla.ScyllaDB
+}
+
+func NewMirrorBadger(conf *config.Config, stats stats.Stats) (*MirrorBadger, error) {
+	badger := badger.NewBadgerDB(conf, stats, badger.DefaultPath())
+	scylla, err := scylla.New(conf, stats)
+	if err != nil {
+		return nil, err
+	}
+	return &MirrorBadger{
+		badger: badger,
+		scylla: scylla,
+	}, nil
+}
+
+func (mb *MirrorBadger) Close() {
+	mb.scylla.Close()
+	mb.badger.Close()
+}
+
+func (mb *MirrorBadger) Get(kv types.KeyValue) (bool, int64, error) {
+	return mb.badger.Get(kv)
```

Comment:
Correct me if I am wrong. our intention here is to populate data in both places(scylla and badger). if we are not doing get on scylla as well, then the in-memory cache in scylla will not be populated so commit will not insert any data in scylla.

we should also add a test to check if data is populated in both places